src/Aspire.Hosting.AWS/Provisioning/Provisioners/AWSProvisioner.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Lifecycle;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Hosting.AWS.Provisioning;
+
+internal sealed class AWSProvisioner(
+    DistributedApplicationExecutionContext executionContext,
+    IServiceProvider serviceProvider,
+    ResourceNotificationService notificationService,
+    ResourceLoggerService loggerService) : IDistributedApplicationLifecycleHook
+{
+    public async Task BeforeStartAsync(DistributedApplicationModel appModel, CancellationToken cancellationToken = default)
+    {
+        if (executionContext.IsPublishMode)
+        {
+            return;
+        }
+
+        var awsResources = appModel.Resources.OfType<IAWSResource>().ToList();
+        if (awsResources.Count == 0)
+        {
+            return;
+        }
+
+        // parent -> children lookup
+        var parentChildLookup = appModel.Resources.OfType<IResourceWithParent>()
```

Comment:
Added a comment to the code.
--
Create a lookup for all resources implementing IResourceWithParent and have IAWSResource as parent in the tree.
Typical childeren that are listed here are ICDKResource, IStackResource with IConstructResource as children. This is important for state reporting so that a stack and it child resources are handled.